src/elements/tabs/tab-label/tab-label.component.ts

```diff
@@ -42,6 +51,129 @@ class SbbTabLabelElement extends SbbDisabledMixin(
+      ? (this.nextElementSibling as SbbTabElement)
+      : null;
+  }
+
+  /** Get the parent `sbb-tab-group`. */
+  public get tabGroup(): SbbTabGroupElement | null {
+    return this.closest('sbb-tab-group');
+  }
+
+  public constructor() {
+    super();
+
+    this.addEventListener('click', () => this.select());
+  }
+
+  public override connectedCallback(): void {
+    super.connectedCallback();
+    this.slot = 'tab-bar';
+    this.tabIndex = this.active ? 0 : -1;
+  }
+
+  protected override willUpdate(changedProperties: PropertyValues<this>): void {
+    super.willUpdate(changedProperties);
+
+    if (changedProperties.has('active')) {
+      this.internals.ariaSelected = `${this.active}`;
+      this.tab?.toggleAttribute('data-active', this.active);
+
+      if (this.active && !this.disabled) {
+        this.select();
+      } else {
+        this.deactivate();
+      }
+    }
+
+    if (changedProperties.has('disabled')) {
+      if (this.disabled) {
+        this.disable();
+      } else {
+        this.enable();
+      }
+    }
+  }
+
+  /** Deactivate the tab. */
+  public deactivate(): void {
+    this.active = false;
+    this._selected = false;
+    this.tabIndex = -1;
+  }
+
+  /** Disable the tab; if it's active, select the first tab in the group. */
+  public disable(): void {
+    if (this.disabled) {
+      return;
+    }
+    this.disabled = true;
+    this.tabIndex = -1;
+    if (this.active) {
+      this.deactivate();
+      this.tabGroup?.activateTab(0);
+    }
+  }
+
+  /** Enable the tab. */
+  public enable(): void {
+    if (this.disabled) {
+      this.disabled = false;
+    }
+  }
+
+  /** Select the tab, deactivating the current selected one, and dispatch the tabchange event. */
+  public select(): void {
+    if (this.disabled) {
+      if (import.meta.env.DEV) {
+        console.warn('You cannot activate a disabled tab');
+      }
+      return;
+    }
+
+    const tabLabels: SbbTabLabelElement[] = this.tabGroup?.tabLabels || [];
+    const prevActiveTabLabel = tabLabels.find((e) => e._selected);
+    if (prevActiveTabLabel !== this) {
+      prevActiveTabLabel?.deactivate();
+      this.active = true;
+      this._selected = true;
+      this.tabIndex = 0;
+      this.dispatchEvent(new Event('active', { bubbles: true, composed: true }));
+      this.tabGroup?.dispatchEvent(
+        new CustomEvent<SbbTabChangedEventDetails>('tabchange', {
+          bubbles: true,
+          composed: true,
+          detail: {
+            activeIndex: tabLabels.findIndex((e) => e === this),
+            activeTabLabel: this,
+            activeTab: this.tab as SbbTabElement,
+            previousIndex: tabLabels.findIndex((e) => e === prevActiveTabLabel),
+            previousTabLabel: prevActiveTabLabel,
+            previousTab: prevActiveTabLabel?.tab as SbbTabElement,
+          },
+        }),
+      );
+    }
+  }
+
+  /**
+   * @internal
+   */
+  public linkToTab(): void {
```

Comment:
we should make it protected, see autocomplete for an example on how to call it ( `this.activeOption?.['selectViaUserInteraction'](true);` )